etc/c/odbc/sqlext.d

```diff
@@ -0,0 +1,2154 @@
+}
+
+// * C datatype to SQL datatype mapping *
+enum
+{
+    SQL_C_DATE      = SQL_DATE,
+    SQL_C_TIME      = SQL_TIME,
+    SQL_C_TIMESTAMP = SQL_TIMESTAMP
+}
+
+enum
+{
+    SQL_C_TYPE_DATE                 = SQL_TYPE_DATE,
+    SQL_C_TYPE_TIME                 = SQL_TYPE_TIME,
+    SQL_C_TYPE_TIMESTAMP            = SQL_TYPE_TIMESTAMP,
+    SQL_C_INTERVAL_YEAR             = SQL_INTERVAL_YEAR,
+    SQL_C_INTERVAL_MONTH            = SQL_INTERVAL_MONTH,
+    SQL_C_INTERVAL_DAY              = SQL_INTERVAL_DAY,
+    SQL_C_INTERVAL_HOUR             = SQL_INTERVAL_HOUR,
+    SQL_C_INTERVAL_MINUTE           = SQL_INTERVAL_MINUTE,
+    SQL_C_INTERVAL_SECOND           = SQL_INTERVAL_SECOND,
+    SQL_C_INTERVAL_YEAR_TO_MONTH    = SQL_INTERVAL_YEAR_TO_MONTH,
+    SQL_C_INTERVAL_DAY_TO_HOUR      = SQL_INTERVAL_DAY_TO_HOUR,
+    SQL_C_INTERVAL_DAY_TO_MINUTE    = SQL_INTERVAL_DAY_TO_MINUTE,
+    SQL_C_INTERVAL_DAY_TO_SECOND    = SQL_INTERVAL_DAY_TO_SECOND,
+    SQL_C_INTERVAL_HOUR_TO_MINUTE   = SQL_INTERVAL_HOUR_TO_MINUTE,
+    SQL_C_INTERVAL_HOUR_TO_SECOND   = SQL_INTERVAL_HOUR_TO_SECOND,
+    SQL_C_INTERVAL_MINUTE_TO_SECOND = SQL_INTERVAL_MINUTE_TO_SECOND
+}
+
+enum
+{
+    SQL_C_BINARY      = SQL_BINARY,
+    SQL_C_BIT         = SQL_BIT,
+    SQL_C_SBIGINT     = (SQL_BIGINT+SQL_SIGNED_OFFSET),      /* SIGNED   BIGINT   */
+    SQL_C_UBIGINT     = (SQL_BIGINT+SQL_UNSIGNED_OFFSET),    /* UNSIGNED BIGINT   */
+    SQL_C_TINYINT     = SQL_TINYINT,
+    SQL_C_SLONG       = (SQL_C_LONG + SQL_SIGNED_OFFSET),    /* SIGNED INTEGER    */
+    SQL_C_SSHORT      = (SQL_C_SHORT + SQL_SIGNED_OFFSET),   /* SIGNED SMALLINT   */
+    SQL_C_STINYINT    = (SQL_TINYINT + SQL_SIGNED_OFFSET),   /* SIGNED TINYINT    */
+    SQL_C_ULONG       = (SQL_C_LONG + SQL_UNSIGNED_OFFSET),  /* UNSIGNED INTEGER  */
+    SQL_C_USHORT      = (SQL_C_SHORT + SQL_UNSIGNED_OFFSET), /* UNSIGNED SMALLINT */
+    SQL_C_UTINYINT    = (SQL_TINYINT + SQL_UNSIGNED_OFFSET), /* UNSIGNED TINYINT  */
+    SQL_C_BOOKMARK    = SQL_C_ULONG,                         /* BOOKMARK          */
+    SQL_C_VARBOOKMARK = SQL_C_BINARY,
+
+    // ODBCVER >= 0x0350
+    SQL_C_GUID        = SQL_GUID                             /* GUID              */
+}
+
+enum int SQL_TYPE_NULL =  0;
+
+// * define for SQL_DIAG_ROW_NUMBER and SQL_DIAG_COLUMN_NUMBER *
+enum : uint
+{
+    SQL_NO_ROW_NUMBER         = (-1),
+    SQL_NO_COLUMN_NUMBER      = (-1),
+    SQL_ROW_NUMBER_UNKNOWN    = (-2),
+    SQL_COLUMN_NUMBER_UNKNOWN = (-2)
+}
+
+// * SQLBindParameter extensions *
+enum uint SQL_DEFAULT_PARAM = (-5);
+enum uint SQL_IGNORE        = (-6);
+enum uint SQL_COLUMN_IGNORE = SQL_IGNORE;
+
+enum : uint
+{
+    SQL_LEN_DATA_AT_EXEC_OFFSET  = (-100),
+}
+
+uint SQL_LEN_DATA_AT_EXEC
+(
+    uint length
+)
+{
+    return ( ( -1 * length ) + cast(uint)SQL_LEN_DATA_AT_EXEC_OFFSET );
+}
+
+// * binary length for driver specific attributes *
+enum uint SQL_LEN_BINARY_ATTR_OFFSET  = (-100);
+
+uint SQL_LEN_BINARY_ATTR
+(
+    uint length
+)

```

Comment:
Very odd syntax for a function declaration. I didn't even recognized it being a function at first.